project/TestParallelization.scala

```diff
@@ -48,7 +48,8 @@ object TestParallelization {
    */
   lazy val simpleGroupingStrategySettings = Seq(
     Test / forkTestJVMCount := {
-      sys.env.get("TEST_PARALLELISM_COUNT").map(_.toInt).getOrElse(4)
+      sys.env.get("TEST_PARALLELISM_COUNT").map(_.toInt)
```

Comment:
we dont need to change this for speeding up github tests.. isnt it?

Reply:
if not, lets separate out user-facing defaults from github tests.
each carry a different risk. increasing parallelism too much can lead to flakiness, and so using all available processors may be problematic.

Reply:
makes sense, I've reverted this for now